aws/resource_aws_security_group_rule.go

```diff
@@ -759,7 +759,19 @@ func setFromIPPerm(d *schema.ResourceData, sg *ec2.SecurityGroup, rule *ec2.IpPe
 		s := rule.UserIdGroupPairs[0]
 
 		if isVPC {
-			d.Set("source_security_group_id", s.GroupId)
+			if existingSourceSgId, ok := d.GetOk("source_security_group_id"); ok {
+				sgIdComponents := strings.Split(existingSourceSgId.(string), "/")
+				hasAccountIdPrefix := len(sgIdComponents) == 2
+
+				if hasAccountIdPrefix {
+					// then ensure on refresh that we prefix the account id
+					d.Set("source_security_group_id", *s.UserId+"/"+*s.GroupId)
```

Comment:
To prevent potential panics (as recommended above), we should use the AWS Go SDK conversion functions, in case the new requirement of `s.UserId` does happen to be `nil` in some environments:

```suggestion
					d.Set("source_security_group_id", fmt.Sprintf("%s/%s", aws.StringValue(s.UserId), aws.StringValue(s.GroupId))
```